lib/publiccloud/utils.pm

```diff
@@ -260,6 +266,20 @@ sub get_ssh_private_key_path {
     return (is_azure() || is_openstack() || get_var('PUBLIC_CLOUD_LTP')) ? "~/.ssh/id_rsa" : '~/.ssh/id_ed25519';
 }
 
+sub permit_root_passwordless {
+    my ($instance) = @_;
+
+    # Permit root passwordless login over SSH
+    $instance->ssh_assert_script_run('sudo cat /etc/ssh/sshd_config');
+    $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
```

Comment:
Let's make it more generic to catch any PermitRootLogin setting.  Also, you can drop the final `g`.  

I think we can also use a drop-in configuration file in /etc/ssh/sshd_config.d to override these settings.

```suggestion
    $instance->ssh_assert_script_run('sudo sed -i "s/^PermitRootLogin .*/PermitRootLogin prohibit-password/" /etc/ssh/sshd_config');
```

Reply:
So basically:
```perl
$instance->ssh_assert_script_run('echo "PermitRootLogin prohibit-password" | sudo tee /etc/ssh/sshd_config.d/001_sshd_permit_passwordless_root.conf');
```
?

Reply:
I think that would work, yes!

Reply:
```
# ssh -tvE /var/tmp/ssh_sut.log  -i '~/.ssh/id_rsa' "azureuser@52.163.64.154" -- $'sudo ls -la /etc/ssh/'; echo 2df~K-$?-
total 620
drwxr-xr-x  2 root root    277 Jul 26 05:42 .
drwxr-xr-x 85 root root   8192 Jul 26 05:43 ..
-rw-------  1 root root 577834 Apr 15 13:53 moduli
-rw-r--r--  1 root root   2403 Apr 15 13:53 ssh_config
-rw-------  1 root root   1426 Jul 26 05:42 ssh_host_dsa_key
-rw-r--r--  1 root root    634 Jul 26 05:42 ssh_host_dsa_key.pub
-rw-------  1 root root    545 Jul 26 05:42 ssh_host_ecdsa_key
-rw-r--r--  1 root root    206 Jul 26 05:42 ssh_host_ecdsa_key.pub
-rw-------  1 root root    444 Jul 26 05:42 ssh_host_ed25519_key
-rw-r--r--  1 root root    126 Jul 26 05:42 ssh_host_ed25519_key.pub
-rw-------  1 root root   2643 Jul 26 05:42 ssh_host_rsa_key
-rw-r--r--  1 root root    598 Jul 26 05:42 ssh_host_rsa_key.pub
-rw-r-----  1 root root   3445 Jul 26 05:42 sshd_config
```

Reply:
This now works. Please recheck.